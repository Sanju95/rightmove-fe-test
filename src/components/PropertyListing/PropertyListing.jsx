import React, { useState, useEffect } from 'react';
import PropertyCard from '../PropertyCard';
import './PropertyListing.scss';

const PropertyListing = () => {
    const [properties, setProperties] = useState([]);

    const fetchProperties = async () => {
        try {
            const res = await fetch('http://localhost:3000/api/properties');
            const date = await res.json();
            setProperties(date);
        } catch (error) {
            console.error('Error fetching properties', error);
        }
    };

    useEffect(() => {
        fetchProperties();
    }, []);

    return (
        <ul className="PropertyListing">
            {properties?.map((property) => (
                <li key={property?.id}>
                    <PropertyCard {...property} />
                </li>
            ))}
        </ul>
    );
};

export default PropertyListing;
