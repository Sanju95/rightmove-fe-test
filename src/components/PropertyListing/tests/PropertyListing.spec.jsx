import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { within } from '@testing-library/dom';
import PropertyListing from '../PropertyListing';

describe('PropertyListing', () => {
    // it('should render five property cards', async () => {
    //     render(<PropertyListing />);
    //     const propertiesList = screen.getByRole('list');
    //     const propertyCards = await within(propertiesList).findAllByRole('listitem');
    //     expect(propertyCards).toHaveLength(5);
    // });

    const mockProperties = [
        {
            id: 1,
            bedrooms: 2,
            summary: 'Property 1',
            price: 100,
        },
        {
            id: 2,
            bedrooms: 3,
            summary: 'Property 2',
            price: 200,
        },
    ];

    it('renders PropertyCards after fetching data', async () => {
        global.fetch = jest.fn().mockResolvedValue({
            json: jest.fn().mockResolvedValue(mockProperties),
        });

        render(<PropertyListing />);

        await waitFor(() => {
            const propertyCards = screen.getAllByRole('listitem');
            expect(propertyCards).toHaveLength(2);
            expect(screen.getByRole('list')).toMatchSnapshot();
        });
    });
});
