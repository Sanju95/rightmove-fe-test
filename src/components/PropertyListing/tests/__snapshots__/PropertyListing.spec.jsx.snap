// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyListing renders PropertyCards after fetching data 1`] = `
<ul
  class="PropertyListing"
>
  <li>
    <div
      class="PropertyCard"
    >
      <div
        class="ImageAndPrice"
      >
        <div
          class="image"
        >
          <img
            alt="Property"
          />
        </div>
        <div
          class="price"
        >
          £ 
          100
        </div>
      </div>
      <div
        class="PropertyDetails"
      >
        <a>
          <h1
            class="propertyTitle"
          />
          <p
            class="displayAddress"
          />
          <p
            class="summary"
          >
            Property 1
          </p>
        </a>
        <div
          class="agentInfo"
        >
          <a
            href="mailto:fakemail.fedtest@rightmove.co.uk"
          >
            <span
              aria-label="Email icon"
              role="img"
            >
              ✉️
            </span>
            Contact 
          </a>
        </div>
      </div>
    </div>
  </li>
  <li>
    <div
      class="PropertyCard"
    >
      <div
        class="ImageAndPrice"
      >
        <div
          class="image"
        >
          <img
            alt="Property"
          />
        </div>
        <div
          class="price"
        >
          £ 
          200
        </div>
      </div>
      <div
        class="PropertyDetails"
      >
        <a>
          <h1
            class="propertyTitle"
          />
          <p
            class="displayAddress"
          />
          <p
            class="summary"
          >
            Property 2
          </p>
        </a>
        <div
          class="agentInfo"
        >
          <a
            href="mailto:fakemail.fedtest@rightmove.co.uk"
          >
            <span
              aria-label="Email icon"
              role="img"
            >
              ✉️
            </span>
            Contact 
          </a>
        </div>
      </div>
    </div>
  </li>
</ul>
`;
